experimental/traffic-portal/src/app/core/cache-groups/divisions/table/divisions-table.component.ts

```diff
@@ -129,7 +131,12 @@ export class DivisionsTableComponent implements OnInit {
 				await this.router.navigate(["/core/division", data.id]);
 				break;
 			case "viewRegions":
-				await this.router.navigate(["/core/regions"]);
+				const region = (await this.regions).find(regionDiv => regionDiv.division === data.id);
+				if (region === undefined) {
+					await this.router.navigate(["/core/regions"]);
+				} else {
+					await this.router.navigate(["/core/region", region.id]);
```

Comment:
So the way this is written, it will take you to a view of some region that belongs to the division, if one exists. If not, it shows you all regions.

But what it _should_ do is take the user to a view of regions that has been filtered to contain only those regions that belong to the division. You shouldn't need to make any API calls in this file to do that.

I'm not sure that the generic table component in tpv2 has been set up with query string parameter filtering yet, so this may not even be possible without extensive work in the generic table component. Therefore, I strongly suggest you don't even try. We can just add an issue to fix it later.

Reply:
Ok, I'll roll these changes back then.